src/components/Form.js

```diff
@@ -1,23 +1,27 @@
 import React, { useState } from 'react';
 import { useDispatch } from 'react-redux';
 import { v4 as uuidv4 } from 'uuid';
-import { addBook } from '../redux/books/booksSlice';
+import { fetchBooks, postBooks } from '../redux/books/booksSlice';
 
 function Form() {
   const [title, setTitle] = useState('');
   const [author, setAuthor] = useState('');
   const [category, setCategory] = useState('');
+
   const dispatch = useDispatch();
 
+  const id = `item${uuidv4()}`;
+
   const handleSubmit = (e) => {
     e.preventDefault();
     const newObj = {
-      id: uuidv4(),
+      item_id: id,
       title,
       author,
       category,
     };
-    dispatch(addBook(newObj));
+
+    dispatch(postBooks(newObj)).then((res) => dispatch(fetchBooks()));
```

Comment:
- [Optional] Instead of making two API calls every time you add/delete a book, kindly consider making just one API call to update the data in the API, and dispatch an action that will update the redux store without making another API call.